### PR TITLE
Kafka operator renaming

### PR DESCRIPTION
**Pull request by fekete-robert**
Renames Kafka operator to Koperator to comply with Apache trademark usage policies

### CODE REVIEW

**Review by baluchicken — APPROVED** [review]

**Review by hi-im-aren — APPROVED** [review]

**Review by stoader — DISMISSED** [review]

**Review by chrisgacsal — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -20,13 +18,13 @@
 
 </p>
 
-# Kafka-Operator
+# Koperator
 
-The Banzai Cloud Kafka operator is a Kubernetes operator to automate provisioning, management, autoscaling and operations of [Apache Kafka](https://kafka.apache.org) clusters deployed to K8s.
+Koperator (formerly called Kafka Operator) is an operator for managing Apache Kafka on Kubernetes that automates the provisioning, management, and autoscaling of [Apache Kafka](https://kafka.apache.org) clusters deployed to K8s.
```
Comment (adamantal) [inline README.md:23]:
```suggestion
Koperator (formerly called Banzai Cloud Kafka Operator) is an operator for managing Apache Kafka on Kubernetes that automates the provisioning, management, and autoscaling of [Apache Kafka](https://kafka.apache.org) clusters deployed to K8s.
```

File: docs/developer.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # Developer Guide
 
-## How to run Kafka-operator in your cluster with your changes
+## How to run Koperator in your cluster with your changes
 
-The Kafka operator is built on the [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) project.
+Koperator (formerly called Kafka Operator) is built on the [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) project.
```
Comment (adamantal) [inline docs/developer.md:5]:
```suggestion
Koperator (formerly called Banzai Cloud Kafka Operator) is built on the [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) project.
```

**Review by adamantal — COMMENTED** [review]
LGTM overall. 

One nit: this operator is commonly referred to as the "Banzai Kafka Operator" by multiple external resources and comparisons of Kafka on Kubernetes solutions as I searched through the internet. Some sources: [[1]](https://cloudhut.dev/blog/2021-06-24-running-kafka-on-kubernetes/), [[2]](https://kai-waehner.medium.com/kafka-operator-for-kubernetes-to-establish-a-cloud-native-apache-kafka-platform-76e893987158), [[3]](https://prune998.medium.com/banzaicloud-kafka-operator-and-broker-autoscaling-1c7324260de1) and [[4]](https://dukemon.leetserve.com/posts/2020/01/banzai-kafka-in-kubernetes/).

I think it'd ease the transition for external users to emphasise the origins (Banzai Cloud) even though it is now officially part of Cisco, and the name of the operator has to be changed due to legal reasons.

**Review by adamantal — APPROVED** [review]
Thanks for making this change!

**Review by hi-im-aren — APPROVED** [review]

**Pull request merged**